src/PackageUrl.cs

```diff
@@ -276,7 +276,7 @@ private string ValidateName(string name)
             }
             return Type switch
             {
-                "nuget" or "cocoapods" or "cpan" or "vsm" or "cran" => name,
+                "nuget" or "cocoapods" or "cpan" or "vsm" or "cran" or "npm" => name,
                 "pypi" => name.Replace('_', '-').ToLower(),
                 _ => name.ToLower()
```

Comment:
Should we align name and namespace validations with python implementation: https://github.com/package-url/packageurl-python/blob/6c96667/src/packageurl/__init__.py#L94? i.e. ToLower() transformation is applied unless we are sure it's necessary.

Reply:
Great point - I'd be happy to contribute that next week, if you don't mind that going in as a separate PR.